spec/spec.md

```diff
@@ -264,12 +265,15 @@ After [session establishment](#session-establishment) completes, the wallet endp
 
 After the [session establishment](#session-establishment) process completes, every message received by an endpoint is expected to be encrypted with AES-128-GCM (as specified by [NIST SP 800-38D](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf)). The sending endpoint should prepare the encrypted message by concatenating:
 
+- a 4-byte message sequence number
 - a random 12-byte IV (which should be newly generated for each encrypted message)
-- the AES-128-GCM message ciphertext, encrypted using the private key created during [session establishment](#session-establishment) and the random IV
+- the AES-128-GCM message ciphertext, encrypted using the private key created during [session establishment](#session-establishment) and the random IV, and including the 4-byte message sequence number as additional authorized data (AAD)
 - the 16-byte authentication tag generated during AES-128-GCM encryption
 
 After decrypting the ciphertext with the shared secret generated during [session establishment](#session-establishment) and verifying the authentication tag, the receiving endpoint should further interpret it as a [JSON-RPC 2.0](https://www.jsonrpc.org/specification) message.
 
+The message sequence number is a monotonically increasing unsigned integer, which starts at 1 when session establishment completes. Each endpoint maintains its own independent sequence number, and increments it each time an encrypted message is created and sent. On receipt of an encrypted message, each endpoint should ensure that the sequence number is greater than that of the previous message received (other than for the first message received). If the value is less than or equal to the previously observed sequence number, the encrypted message should be discarded and the connection closed.
```

Comment:
This provides resistance to replay attacks

Reply:
I commented on #112, but: why less than or equal to? Why not ‘exactly the next number in sequence?’

Reply:
Left my comments on this topic here: https://github.com/solana-mobile/mobile-wallet-adapter/pull/112#discussion_r922849562